library/pkparse.c

```diff
@@ -446,6 +446,149 @@ static int pk_group_id_from_specified(const mbedtls_asn1_buf *params,
+/*
+ * Create a temporary ecp_keypair for converting an EC point in compressed
+ * format to an uncompressed one
+ */
+static int pk_convert_compressed_ec(mbedtls_pk_context *pk,
+                                    unsigned char *in_start, size_t in_len,
+                                    size_t *out_buf_len, unsigned char *out_buf,
+                                    size_t out_buf_size)
+{
+    mbedtls_ecp_keypair ecp_key;
+    mbedtls_ecp_group_id ecp_group_id;
+    int ret;
+
+    ecp_group_id = mbedtls_ecc_group_of_psa(pk->pk_ec_family, pk->pk_bits, 0);
+
+    mbedtls_ecp_keypair_init(&ecp_key);
+    ret = mbedtls_ecp_group_load(&(ecp_key.grp), ecp_group_id);
+    if (ret != 0) {
+        return ret;
+    }
+    ret = mbedtls_ecp_point_read_binary(&(ecp_key.grp), &ecp_key.Q,
+                                        in_start, in_len);
+    if (ret != 0) {
+        mbedtls_ecp_keypair_free(&ecp_key);
+        return ret;
+    }
+    ret = mbedtls_ecp_point_write_binary(&(ecp_key.grp), &ecp_key.Q,
+                                         MBEDTLS_ECP_PF_UNCOMPRESSED,
+                                         out_buf_len, out_buf, out_buf_size);
+    if (ret != 0) {
+        mbedtls_ecp_keypair_free(&ecp_key);
+        return ret;
+    }
+    mbedtls_ecp_keypair_free(&ecp_key);
+
+    return 0;
+}
+
+static int pk_get_ecpubkey(unsigned char **p, const unsigned char *end,
+                           mbedtls_pk_context *pk)
+{
+    mbedtls_svc_key_id_t key;
+    psa_key_attributes_t key_attrs = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status;
+    size_t len = (end - *p);
+    int ret;
+
+    if ((len > PSA_EXPORT_PUBLIC_KEY_MAX_SIZE) ||
+        (*p == NULL) || (end == NULL) ||
+        (pk == NULL)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+
+    /* Compressed point format are not supported yet by PSA crypto. As a
+     * consequence ecp functions are used to "convert" the point to
+     * uncompressed format */
+    if ((**p == 0x02) || (**p == 0x03)) {
+        ret = pk_convert_compressed_ec(pk, *p, len,
+                                       &(pk->pk_raw_len), pk->pk_raw,
+                                       PSA_EXPORT_PUBLIC_KEY_MAX_SIZE);
+        if (ret != 0) {
+            return ret;
+        }
+    } else {
+        /* Uncompressed format */
+        memcpy(pk->pk_raw, *p, (end - *p));
+        pk->pk_raw_len = end - *p;
+    }
+
+    /* Validate the key by trying to importing it */
+    psa_set_key_usage_flags(&key_attrs, 0);
+    psa_set_key_algorithm(&key_attrs, PSA_ALG_ECDSA_ANY);
+    psa_set_key_type(&key_attrs, PSA_KEY_TYPE_ECC_PUBLIC_KEY(pk->pk_ec_family));
+    psa_set_key_bits(&key_attrs, pk->pk_bits);
+
+    status = psa_import_key(&key_attrs, pk->pk_raw, pk->pk_raw_len, &key);
+    psa_destroy_key(key);
+    if (status != PSA_SUCCESS) {
+        mbedtls_platform_zeroize(pk->pk_raw, MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN);
+        pk->pk_raw_len = 0;
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
```

Comment:
Just to ensure. Do we need to verify if the key is valid by importin it here?

The status of `psa_destroy_key` is not verified. I think it should be checked also.

Reply:
Since the "original" function did `mbedtls_ecp_point_read_binary` + `mbedtls_ecp_check_pubkey` I thought that adding a check also in the "new" one was the best way to have symmetry between the 2 versions. Am I wrong?

PS = I will add the check for `psa_destroy_key` ;)

Reply:
> Since the "original" function did mbedtls_ecp_point_read_binary + mbedtls_ecp_check_pubkey I thought that adding a check also in the "new" one was the best way to have symmetry between the 2 versions. Am I wrong?

Yes, in this case it make sense.